src/ciel/plugin.go

```diff
@@ -0,0 +1,59 @@
+package main
+
+import (
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"syscall"
+)
+
+const (
+	PluginDir = LibExecDir + "/ciel-plugin/"
+)
+
+// Plugin defines a ciel plugin
+type Plugin struct {
+	Name string
+	// TODO: get script usage from header comments
+	Usage string
+}
+
+func cielPlugin() int {
+	proc := PluginDir + "ciel-" + SubCommand
+	cmd := exec.Command(proc, SubArgs...)
+	cmd.Stdin = os.Stdin
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	err := cmd.Run()
+	if err != nil {
+		if exitError, ok := err.(*exec.ExitError); ok {
+			exitStatus := exitError.Sys().(syscall.WaitStatus)
+			return exitStatus.ExitStatus()
+		}
+		log.Printf("failed to run plugin %s: %v\n", SubCommand, err)
+		return 1
+	}
+	return 0
+}
+
+func getPlugins() []Plugin {
+	var Plugins []Plugin
+	err := filepath.Walk(PluginDir, func(path string, f os.FileInfo, err error) error {
+		if f == nil {
+			return err
+		}
+		if f.IsDir() {
+			return nil
```

Comment:
That's exactly the source for "accepting all plugins place at sub-directories"
Skipping `f.IsDir()` is just avoiding processing directories, the files under that directory will still be walked through.

A simple case would be 
```bash
everette@LER0ever-T460P-Manjaro /u/l/ciel-plugin> tree
.
├── adir
│   └── ciel-goodmorning
├── ciel-build
├── ciel-generate
├── ciel-merge-cache
├── ciel-release
├── ciel-update
└── ciel-~update

1 directory, 7 files
everette@LER0ever-T460P-Manjaro /u/l/ciel-plugin> ciel help
Usage: ciel [command [...]]
Default command is "shell".

Commands:
	help
	version

	init   <tarball>

	drop   [<layers>]
	mount  [--read-write] [<layers>]
	merge  [<upper>..]<lower> [--no-self] path
	clean  [--factory-reset]

	shell  [<cmdline>]
	rawcmd <cmd> <arg1> <arg2> ...

Plugins:
	goodmorning
	build
	generate
	merge-cache
	release
	update
	~update
```